### DISCUSSION TITLE
Implicitly typed constants

### DISCUSSION THREAD

**Discussion by ekolis** [dotnet/csharplang#4863] [Language Ideas]
I can say:

`var Age = 18;`

However I can't say:

`const Age = 18;`

I have to specify the type of the constant explicitly:

`const int Age = 18;`

How about allowing for implicitly typed constants so that the type specification can be optional?

**Comment by huoyaoyuan** [comment]
This is something the language team avoiding. A numeric literal doesn't points its type obviously. The same rule for fields applies to constants (requiring explicit type).

**Comment by 333fred** [comment]
See https://github.com/dotnet/csharplang/issues/106.
